### PR TITLE
Delete archived course students + ToR cache clear

### LINKED ISSUE #2694: Archived course students in Pyramus are not removed in Muikku

### PR DESCRIPTION
**Pull request by pkukkon**
- fixes #2694

### CODE REVIEW

File: muikku-school-data-pyramus/src/main/java/fi/otavanopisto/muikku/plugins/schooldatapyramus/PyramusUpdater.java [hunk 1]
```diff
@@ -628,6 +628,15 @@ public boolean updateCourseStudent(Long courseStudentId, Long courseId, Long stu
     
     return false;
   }
+  
+  public void removeCourseStudent(Long courseStudentId, Long courseId, Long studentId) {
+    String identifier = identifierMapper.getWorkspaceStudentIdentifier(courseStudentId);
+    SchoolDataIdentifier workspaceUserIdentifier = new SchoolDataIdentifier(identifier, SchoolDataPyramusPluginDescriptor.SCHOOL_DATA_SOURCE);
+    WorkspaceUserEntity workspaceUserEntity = workspaceUserEntityController.findWorkspaceUserEntityByWorkspaceUserIdentifierIncludeArchived(workspaceUserIdentifier);
+    if (workspaceUserEntity != null) {
+      workspaceUserEntityController.deleteWorkspaceUserEntity(workspaceUserEntity);
```
Comment (anttiviljakainen) [inline muikku-school-data-pyramus/src/main/java/fi/otavanopisto/muikku/plugins/schooldatapyramus/PyramusUpdater.java]: Is it safe? :P I mean, are there dependencies at all pointing to this and what are the effects if there are?

Comment (pkukkon) [inline muikku-school-data-pyramus/src/main/java/fi/otavanopisto/muikku/plugins/schooldatapyramus/PyramusUpdater.java]: Admittedly not 100% safe. References WorkspaceUserPermissionOverride, which has never been used so far, and AssessmentRequestMessageId, which is used in code but seems to serve no actual reason in the application itself (that, or the functionality planned to make use of that information has not yet been implemented, whatever and whenever that would be).

Comment (anttiviljakainen) [inline muikku-school-data-pyramus/src/main/java/fi/otavanopisto/muikku/plugins/schooldatapyramus/PyramusUpdater.java]: It's deleting WorkspaceUserEntity, I was just wondering if anything has dependencies (like answers on tasks etc) on it, can it cause some sort of headache?

**Comment by anttiviljakainen** [comment]
Requires another solution (use archived to mark this as archived and make another flag to mark the entity as "archived from [ie failed/passed] course")

**Pull request closed without merging**